### PR TITLE
this ticket has 2 mentioned use cases but only 1 is currently working…

### PR DESCRIPTION
**Pull request by darlp5589**
@chriscodev , kindly review this PR.

Here's the ASANA Link: https://app.asana.com/0/1201943683850147/1202187061972897

There are 2 mentioned use cases here but I want you to check only the first one since the other one is currently under development. 

Here's how you can test: 

1. Add a Logical Network -> Add a new ATSC Channel -> Make inputs/changes -> OK -> Update the same ATSC Channel -> Make modifications → OK

### CODE REVIEW

File: src/app/pages/dashboard/network-config/networks/modal-networks/modal-networks-atsc-channel/modal-networks-atsc-channel.component.ts [hunk 1]
```diff
@@ -50,16 +50,22 @@ export class ModalNetworksAtscChannelComponent implements OnInit, OnChanges {
 		private ds: DataService,
 		private ts: TransportService
 	) {
+		console.log("constructor for Channel: ",this.modalChannelData)
```
Comment (chriscodev) [inline src/app/pages/dashboard/network-config/networks/modal-networks/modal-networks-atsc-channel/modal-networks-atsc-channel.component.ts:53]: remove console.log

File: src/app/pages/dashboard/network-config/networks/networks.component.ts [hunk 2]
```diff
@@ -469,8 +471,14 @@ export class NetworksComponent implements OnInit {
...
 	ATSCChangedHandler($event) {
 		console.log('ATSCChangedHandler', $event)
-		if($event.source === 'DescCallFromNetwork') this.descDataObject = $event; // this will go to Descriptors component
-		else this.channelDataObject = $event;
+		if($event.source === 'DescCallFromNetwork') {
+			this.descDataObject = $event; // this will go to Descriptors component
+		} 
+		else {
+			this.channelDataObject = $event;
+			//if($event.hasOwnProperty('modulation')) this.channelChangedHandler($event);
```
Comment (chriscodev) [inline src/app/pages/dashboard/network-config/networks/networks.component.ts:479]: remove commented code

**Review by chriscodev — CHANGES_REQUESTED** [review]
- remove console logs

- remove commented lines of code

- on add  atsc services table gets destroyed, it should not be destroyed on adding channel settings

crop

File: src/app/pages/dashboard/network-config/networks/networks.component.ts (hunk 2 above)
Comment (darlp5589) [reply src/app/pages/dashboard/network-config/networks/networks.component.ts:479]: done

**Comment by darlp5589** [comment]
@chriscodev , kindly recheck these concerns you sent above. For the service concern, it's not under my ticket scope yet but I will consider that in the future. For the comments and console.logs, I removed some and retained some since they are very crucial on my implementation especially for data passing between components.

In this new commit, kindly check below scenario:
Select "Update" on a server-existing Logical Network -> Update an Existing ATSC Channel → Make modifications → Ok

Just some notes: I haven't implemented the scenario wherein you add a new network, add some channels on that network and include the network in the network data table and edit that newly added network's set of channels. It will be on a new ticket.

**Review by chriscodev — COMMENTED** [review]
atscneterr
atscneterr2  

image1 - datatable initialisation error
image2 - select country is available and yet checkmark is greyed out and ok button is disabled

**Review by chriscodev — APPROVED** [review]
This is approve on my end

**Comment by triveni-codev** [comment]
Merging to master. Implementation structure confirmed for the most part. Kindly take matter on the temporarily replaced implementation of the timezone portion. The timezone has to have the original search function.


~~~~~~
1.Networks Section Development: Accommodation of "Update ATSC Channel" use cases
https://app.asana.com/0/1202172889691467/1202187061972897/f

2.Networks Section Development: Newly added Logical Network(locally dirty all throughout) -> Update existing ATSC Channel 
https://app.asana.com/0/1201468220173184/1202210153506903/f

3.Networks Section - ATSC Channel continuous development: Complete funneling of Added/Updated changes down to the root
https://app.asana.com/0/1202172889691467/1202249852885595/f

**Pull request merged**